consul/consul.go

```diff
@@ -349,6 +349,31 @@ func (c *Consul) marathonToConsulChecks(task *apps.Task, healthChecks []apps.Hea
 	return checks
 }
 
+func (c *Consul) filterMarathonChecks(healthChecks []apps.HealthCheck) []apps.HealthCheck {
+	ignoredTypes := c.getIgnoredHealthCheckTypes()
+	if len(ignoredTypes) == 0 {
+		return healthChecks
+	}
+	var filteredChecks []apps.HealthCheck
+	for _, check := range healthChecks {
+		if !contains(ignoredTypes, check.Protocol) {
+			filteredChecks = append(filteredChecks, check)
+		}
+	}
+	return filteredChecks
+}
+
+func (c *Consul) getIgnoredHealthCheckTypes() []string {
+	ignoredTypes := make([]string, 0)
```

Comment:
OK, I've fixed it so `ToUpper` is called once, but still I'd like to check if the input is valid (trim, remove empty values):
```go
ignoredTypes := make([]string, 0)
	for _, ignoredType := range strings.Split(strings.ToUpper(c.config.IgnoredHealthChecks), ",") {
		var ignoredType = strings.TrimSpace(ignoredType)
		if ignoredType != "" {
			ignoredTypes = append(ignoredTypes, ignoredType)
		}
	}
	return ignoredTypes
```
By default it returns empty splice.